packages/std/src/results/cosmos_msg.rs

```diff
@@ -19,14 +19,29 @@ where
     // to call into more app-specific code (whatever they define)
     Custom(T),
     Staking(StakingMsg),
+    Wasm(WasmMsg),
     /// A Stargate message encoded the same way as a protobof [Any](https://github.com/protocolbuffers/protobuf/blob/master/src/google/protobuf/any.proto).
     /// This is the same structure as messages in `TxBody` from [ADR-020](https://github.com/cosmos/cosmos-sdk/blob/master/docs/architecture/adr-020-protobuf-transaction-encoding.md)
     #[cfg(feature = "stargate")]
     Stargate {
         type_url: String,
         data: Binary,
     },
-    Wasm(WasmMsg),
+    /// Sends *native tokens* owned by the contract to the given address on another chain.
+    /// The channel must already be established between the ibctransfer module on this chain
+    /// and a matching module on the remote chain.
+    /// We cannot select the port_id, this is whatever the local chain has bound the ibctransfer
+    /// module to.
+    #[cfg(feature = "stargate")]
+    IbcSend {
+        /// exisiting channel to send the tokens over
+        channel_id: String,
+        /// address on the remote chain to receive these tokens
+        to_address: HumanAddr,
+        /// packet data only supports one coin
+        /// https://github.com/cosmos/cosmos-sdk/blob/master/proto/ibc/applications/transfer/v1/transfer.proto#L11-L20
+        amount: Coin,
+    },
```

Comment:
Can we have an optional field `port_id:String`? 
The default should be `transfer` but this can be set in the genesis for every chain

Reply:
Optinal fields are not super convenient in Rust (you cannot omit them like in JavaScript/Go). Instead you always have to write them either `port: Some(value)` or `port: None`.

Would it be an option to store this as a constant in the chain code, such that it is added when the Rust type is translated to a Cosmos SDK type?

Reply:
> Can we have an optional field port_id:String?
> The default should be `transfer` but this can be set in the genesis for every chain

This detail is super important for the Go module handling the transfers. There should be one official IbcTransfer port for the application, which may be name `transfer` or `ibctransfer` or `foobar` or `wasm12334556775fg4g43tuhti44tt4yt74yt47`.

The contract should run the same on any chain, regardless of how genesis was set up. By just sending a token, we leave the port naming up to the wasmd-based app. The channel is needed to differentiate various chains and should be set by the calling user.

Does that make sense? Setting a port to anything other than what the ibctransfer module was bound to will lead to an error, so let's just set it for them, as there is only one "right" value